app/src/main/kotlin/compiler/syntax/AstFactory.kt

```diff
@@ -360,6 +360,22 @@ object AstFactory {
                 val bodyBlock = processMaybeBlock(children[4], diagnostics)
                 Statement.Loop(conditionExpr, bodyBlock, combineLocations(children))
             }
+            in listOf(Productions.nonBraceStatementForEach, Productions.nonIfNonBraceStatementForEach) -> {
+                val receivingVariable = Variable(
+                    Variable.Kind.VALUE,
+                    (children[1] as ParseTree.Leaf).content,
+                    processType(children[3]),
+                    null,
+                    combineLocations(children.subList(1, 3)),
+                )
+                val generatorCall = Expression.FunctionCall(
+                    (children[5] as ParseTree.Leaf).content,
+                    processFunctionCallArguments(children[7], diagnostics),
+                    combineLocations(children.subList(5, 7)),
```

Comment:
Could you double check if locations are computed correctly? In Kotlin, `toIndex` in `List.subList` is exclusive and we also need to account closing brace, so imho it should be `(5,9)`.

Reply:
You're right